### PR TITLE
chore: set ZAC Docker Image in Docker Compose file back to 'latest'

### PR DESCRIPTION
**Pull request by edgarvonk**
Set ZAC Docker Image in Docker Compose file back to 'latest' since we do not want to pin this.

Solves PZ-4339

### CODE REVIEW

**Review by niorg — CHANGES_REQUESTED** [review]
Hard disagree. To what image do you want to refer with 'latest'? Whatever image you have locally cached with that tag? That can be months old even. You'll have no clue or control on what will be started, unless you use force pull flags, which is an ugly workaround anyway.

I'd suggest to keep following the most recent released images here, but have renovate automerge those PR's, so we don't need to manually review those.

**Comment by edgarvonk** [comment]
> Hard disagree. To what image do you want to refer with 'latest'? Whatever image you have locally cached with that tag? That can be months old even. You'll have no clue or control on what will be started, unless you use force pull flags, which is an ugly workaround anyway.
> 
> I'd suggest to keep following the most recent released images here, but have renovate automerge those PR's, so we don't need to manually review those.

We do indeed force pulls for local development. See our `start-docker-compose.sh` script. So you always by default use the latest published version. If we pin versions instead you may be running behind a day since currently I think Renovate only creates these PRs nightly? And indeed, then we should use auto merge which we don't have yet.

It's also a lot of overhead, having Renovate constantly create PRs to update the ZAC Docker image tag because typically we release multiple versions of ZAC each day..

**Comment by edgarvonk** [comment]
> > Hard disagree. To what image do you want to refer with 'latest'? Whatever image you have locally cached with that tag? That can be months old even. You'll have no clue or control on what will be started, unless you use force pull flags, which is an ugly workaround anyway.
> > I'd suggest to keep following the most recent released images here, but have renovate automerge those PR's, so we don't need to manually review those.
> 
> We do indeed force pulls for local development. See our `start-docker-compose.sh` script. So you always by default use the latest published version. If we pin versions instead you may be running behind a day since currently I think Renovate only creates these PRs nightly? And indeed, then we should use auto merge which we don't have yet.
> 
> It's also a lot of overhead, having Renovate constantly create PRs to update the ZAC Docker image tag because typically we release multiple versions of ZAC each day..

To be honest, I personally never use this functionality where you run the latest version of ZAC locally as part of your Docker Compose. I always run ZAC in my IDE or I build the ZAC Docker image locally and then use that. But still, I think it is nice functionality to offer (also for our future open source community) e.g. for frontend developers or just for local test and demo purposes. And I think using the 'latest' tag is a good use case for this scenario (and also only for this scenario..) and better then using a pinned image version because of the reasons mentioned above.

**Comment by niorg** [comment]
Just setting it to `latest` won't be helpful either, as renovate will digest-pin that too. So it will only be truly 'updated' once a month.  We would have to add a package rule to explicitly disable that for this image:

```
  "packageRules": [{
    "matchPackageNames": ["ghcr.io/infonl/zaakafhandelcomponent"],
    "matchDatasources": ["docker"],
    "updateTypes": ["digest"],
    "enabled": false
  }]
```

Again, I'd advise to refrain from using `latest` at all. But with above package rule it's back to original behaviour.

**Comment by edgarvonk** [comment]
> Again, I'd advise to refrain from using latest at all. But with above package rule it's back to original behaviour.

Normally I would agree never to use the 'latest' tag and we had a similar discussion back when we implemented this, but still I think in this use case (it is only used for local development and only if you want to use the latest ZAC version) it does make sense. And I do think the downsides I mentioned to the current solution are quite large:
- it causes a lot of overhead since we create new ZAC versions multiple times per day
- unless we create these PRs on an hourly basis you may then run behind a day if you want to use the latest ZAC image locally. how do you propose to solve this in the current solution?

**Comment by edgarvonk** [comment]
> Again, I'd advise to refrain from using `latest` at all. But with above package rule it's back to original behaviour.
One more thing: in the current situation we run our entire build process for every PR (again, there will be a lot of these) where just the ZAC Docker image is updated in our Docker Compose. This serves no purpose at all (since this ZAC Docker image version is only used for local development and for nothing else) and is quite a waste of resources on GitHub. If only for that I am not in favour of this solution and I propose to go back to using the `latest` tag instead.

**Comment by edgarvonk** [comment]
> We would have to add a package rule to explicitly disable that for this image:

But I already made such a package rule to disable updates for ZAC in this PR right?

File: renovate.json [hunk 1]
```diff
@@ -42,6 +43,12 @@
       "matchPackageNames": ["openzaak/open-zaak"],
       "matchDatasources": ["docker"],
       "allowedVersions": "<= 1.15.0"
+    },
+    {
+      "matchPackageNames": ["infonl/zaakafhandelcomponent"],
```
Comment (niorg) [inline renovate.json:48]: The matchPackageNames would need to be `ghcr.io/infonl/zaakafhandelcomponent` , as the package name is derived from the registry (with docker hub being the default). 
Maybe also add a filter for the digest type? Like
`"updateTypes": ["digest"],`

**Review by niorg — CHANGES_REQUESTED** [review]

File: renovate.json (hunk 1 above)
Comment (edgarvonk) [reply renovate.json]: clear. thanks! I updated it.

**Review by infojohn — APPROVED** [review]

**Review by niorg — APPROVED** [review]

**Pull request merged**